### DISCUSSION TITLE
Device support release cycle for Frameit

### DISCUSSION THREAD

**Discussion by ryanhanks-bestow** [fastlane/fastlane#22279] [General] [answered]
What's the expectation here in terms of devices that will be supported?

In particular, let's talk about iPhone.

iPhone 16 was released this month. I learned from https://github.com/jamesjingyi/mockup-device-frames?tab=readme-ov-file#device-mockups that apple publishes these.

I don't feel comfortable signing on to use a tool that's designed specifically for creating these types of device-images when the current release is a full two-years behind the release cycle of the most popular line of mobile devices on the market.

What's the root issue here?

I see how it takes no more time to integrate for these immediately as they're released than it does waiting a year or two.

Is this an issue of bandwidth?

Combing through the issues and I see plenty of discussion around this and users going to lengths to hack-in their relevant accommodations.

What will it take to get the latest lineup of iPhone devices supported by frameit? Is there expectations that the community is to contribute these? Is there documentation on the process for doing such contributions?

@joshdholtz can you please comment on the current state of things? I see you were involved in the release that introduced the iPhone 14 line of devices (https://github.com/fastlane/frameit-frames/pull/31), let me know if there's someone else I can bring into this disucssion.

Thanks

**Comment by sgrgrsn** [comment]
It [looks like](https://github.com/fastlane/fastlane/tree/master/frameit/frames_generator) fastlane is getting the frames from the [Facebook frameset](https://design.facebook.com/toolsandresources/devices/). That catalog has not been updated with either iPhone 15 or 16, so that might be the reason.

So maybe getting them from the repo you linked to is a better idea. I don't know if there are any legal considerations to make though.

**Comment by iBotPeaches** [comment, marked as answer]
There is no good answer and years later - does it matter? fastlane is naturally huge with 19 top level tools that are insanely packed with features. When this was a paid supported tool under Google or Twitter - it was possible to manage with a few engineers.

That said - when funding went bye bye in 2021 - does any engineer really have the free time to do anything at the scale this needed? Especially at $0?

All of that to say - I started working on frameit briefly and added lots of the newer devices. The root issues are:

1. Facebook Frames are gone. This meant the scripts we had to mark frames as legacy and support new ones no longer worked. So what do people do? That is where we are at now. People are just manually adding frames and the legal boundary of where those comes from is a massive gray area.

2. We have no money.

I'll close this out. The expectation for any new device can be summarized as - "When someone needs that support and is willing to put the time forward to add it."

For any Googlers

1. PR this - https://github.com/fastlane/frameit-frames
2. Add device type - https://github.com/fastlane/fastlane/blob/master/frameit/lib/frameit/device_types.rb

**Reply by ryanhanks-bestow** [reply]
Thanks
